### PR TITLE
Tunnel/v8

### PR DESCRIPTION
**Pull request by victorjulien**
Tunnel handling updates to improve thread safety.

Misc related cleanups.

Replaces #9985:
- fixups
- rebase master

### CODE REVIEW

File: src/decode.h [hunk 1]
```diff
@@ -1015,7 +1018,7 @@ void DecodeUnregisterCounters(void);
 #define PKT_STREAM_NOPCAPLOG BIT_U32(12)
 
 #define PKT_TUNNEL           BIT_U32(13)
-#define PKT_TUNNEL_VERDICTED BIT_U32(14)
+// vacancy
 
 /** Packet checksum is not computed (TX packet for example) */
 #define PKT_IGNORE_CHECKSUM BIT_U32(15)
```
Comment (lukashino) [inline src/decode.h:1034]: Can't all the defines be shifted by two?

File: src/decode.h [hunk 2]
```diff
@@ -1135,23 +1155,10 @@ static inline bool PacketIsNotTunnel(const Packet *p)
...
-        // verdict
-        SCLogDebug("root %p: verdict", p);
-    } else if (PacketIsTunnelChild(p) && outstanding == 1 && p->root &&
-               PacketTunnelIsVerdicted(p->root)) {
-        // verdict
-        SCLogDebug("tunnel %p: verdict", p);
-    } else {
-        verdict = false;
-    }
+    verdict = VerdictTunnelPacketInternal(p);
```
Comment (lukashino) [inline src/decode.h:1162]: declare variable here?

File: src/decode.h [hunk 3]
```diff
@@ -1021,8 +1021,9 @@ void DecodeUnregisterCounters(void);
 /** Packet is modified by the stream engine, we need to recalc the csum and       \
                    reinject/replace */
 #define PKT_STREAM_MODIFIED BIT_U32(10)
-/** Packet mark is modified */
-#define PKT_MARK_MODIFIED BIT_U32(11)
+
+// vacancy
```
Comment (lukashino) [inline src/decode.h:1025]: shift here as well?

**Review by lukashino — CHANGES_REQUESTED** [review]

File: src/decode.h (hunk 2 above)
Comment (victorjulien) [reply src/decode.h:1162]: I guess I don't like to declare a var inside a lock, then use it outside. It's a matter of style I think, I don't think it has practical effect

File: src/decode.h (hunk 1 above)
Comment (victorjulien) [reply src/decode.h:1034]: Not shifting to keep flags stable. This mostly helps in debugging reported core dumps.

File: src/decode.h (hunk 2 above)
Comment (lukashino) [reply src/decode.h:1162]: ok, the comment was based on the experience from the previous PRs.

**Review by lukashino — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #10629, thanks!

**Pull request closed without merging**